contracts/ERC4626BufferedUpgradeable.sol

```diff
@@ -312,13 +312,16 @@ abstract contract ERC4626BufferedUpgradeable is Initializable, ERC20Upgradeable,
     }
 
     /// @notice Returns the maximum amount of the underlying asset that can be withdrawn from the owner balance in the
-    /// vault, through a withdraw call. Accounts for shares burned in the reward buffer, but doesn't preview holdings
-    /// state update.
+    /// vault, through a withdraw call.
     /// If `owner` is `protocolFeeReceiver`, this function might underestimate when there are pending protocol fees.
     /// In that case, call the `updateHoldingsState()` right before calling this function,
     /// to ensure that the value of `maxWithdraw` is exact.
     function maxWithdraw(address owner) public view virtual returns (uint256) {
-        return convertToAssets(balanceOf(owner));
+        try ERC4626BufferedUpgradeable(address(this)).previewRedeem(balanceOf(owner)) returns (uint256 amount) {
```

Comment:
Calling `previewRedeem` there is a little confusing to me (although it looks correct because `previewRedeem` is non-virtual). 
However, how about adding a comment like: _Try calling `_previewUpdateHoldingsState` and then `convertToAssets(balanceOf(owner))` on the updated state. So happens that `previewRedeem` does exactly the thing_